### PR TITLE
Panic if endpoint or service is empty in NewNexusClient

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

File: internal/workflow.go [hunk 1]
```diff
@@ -2401,6 +2401,12 @@ type nexusClient struct {
 //
 // NOTE: Experimental
 func NewNexusClient(endpoint, service string) NexusClient {
+	if endpoint == "" {
```
Comment (Quinn-With-Two-Ns) [inline internal/workflow.go:2404]: Just as a sanity check the server does not allow an empty endpoint name to be registered or an empty service right?

**Review by cretz — APPROVED** [review]

File: internal/workflow.go (hunk 1 above)
Comment (bergundy) [reply internal/workflow.go:2404]: Yes, we verify that.

**Review by Quinn-With-Two-Ns — APPROVED** [review]

File: internal/workflow.go [hunk 2]
```diff
@@ -2401,6 +2401,12 @@ type nexusClient struct {
...
+		panic("endpoint must not be empty")
+	}
+	if service == "" {
+		panic("service must not be empty")
+	}
 	return nexusClient{endpoint, service}
```
Comment (Quinn-With-Two-Ns) [inline internal/workflow.go:2410]: You could argue this validation should be post-interceptor, but I am fine with it as is for now

Comment (bergundy) [reply internal/workflow.go:2417]: Hmm... yeah, potentially, but I think getting the line number in the stack trace is better for debugging.

**Pull request merged**